src/planning/src/planning/cone_coloring.cpp

```diff
@@ -0,0 +1,137 @@
+#include "planning/cone_coloring.hpp"
+#include <iostream>
+
+ConeColoring::ConeColoring(double gain_angle, double gain_distance, double gain_ncones,
+ double exponent_1, double exponent_2, double cost_max ) : angle_gain(gain_angle),
+ distance_gain(gain_distance), ncones_gain(gain_ncones), exponent1(exponent_1),
+ exponent2(exponent_2), max_cost(cost_max) {}
+
+double ConeColoring::angle_and_norm(Cone *possible_next_cone, bool side, double &norm) {
```

Comment:
Don't just return one of the parts by argument and the other by return value, use a std::pair or something